rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -1723,6 +1743,56 @@ where
                 )
             })
     }
+
+    // TODO[NNS1-2997]: Remove this function once the migration is completed.
+    pub async fn populate_wasm_metadata(
+        thread_safe_sns: &'static LocalKey<RefCell<SnsWasmCanister<M>>>,
+    ) {
+        // 2. Compute metadata for all marked WASMs.
+        let all_wasm_hashes = thread_safe_sns.with(|sns_canister| {
+            sns_canister
+                .borrow()
+                .wasm_indexes
+                .keys()
+                .cloned()
+                .collect::<Vec<_>>()
+        });
+        for hash in all_wasm_hashes {
+            ic_cdk_timers::set_timer(std::time::Duration::from_millis(0), move || {
```

Comment:
Is it safe to use timers when we haven't migrated from dfn_core?

Reply:
AFAIK this is safe as we're not mixing IC CDK and dfn_core calls. But we probably won't need it in this PR after all, if we go with a sync migration approach.